md2conf/converter.py

```diff
@@ -289,11 +289,20 @@ def __init__(self, path: str, options: ConfluenceDocumentOptions) -> None:
             r"<!--\s+confluence-space-key:\s*(\w+)\s+-->", html
         )
 
+        # extract 'generated-by' tag text
+        generated_by_tag, html = _extract_value(
+            r"<!--\s+generated-by:\s*(.*)\s+-->", html
+        )
+
         # parse Markdown document
         if self.options.generated_by:
+            generated_by = self.options.generated_by
```

Comment:
Can you double-check if the command-line option `--no-generated-by` still works after these changes?

Reply:
I did and it works. If `--no-generated-by` is provided (and only then), `self.options.generated_by` becomes `False` because of `action=store_false`. Otherwise it is always a string, either default or user-defined, so this if-clause becomes `True`.